qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -925,7 +923,7 @@ public void testSnapshotRestore() throws IOException {
         // We therefore use the deprecated typed APIs when running against the current version.
         if (isRunningAgainstOldCluster() == false) {
             createTemplateRequest.addParameter(INCLUDE_TYPE_NAME_PARAMETER, "true");
-            createTemplateRequest.setOptions(expectWarnings(RestPutIndexTemplateAction.TYPES_DEPRECATION_MESSAGE));
+            createTemplateRequest.setOptions(expectTypeRemovalWarnings());
```

Comment:
I think these lines need to be pulled out of this `if` statement, so that warnings are allowed even if we're running against an older cluster.